### ISSUE TITLE
Crash on viewing Orders (NullPointerException Parameter specified as non-null is null)

### ISSUE THREAD

**Issue by thabotswana** [woocommerce/woocommerce-android#6013]
A user reported that the WooAndroid app crashes when accessing orders. They say its working on iOS.

Reported in 4844740-zen.

Sentry reports found [here](https://sentry.io/organizations/a8c/discover/results/?field=title&field=event.type&field=project&field=user.display&field=timestamp&name=All+Events&project=1459556&query=user.email%3Anisan241011%40gmail.com&sort=-timestamp&statsPeriod=7d&yAxis=count%28%29).

**Mobile Environment**
Please include:
- Device: Samsung Galaxy Note10+
- Android version: Android 12
- WooCommerce Android version: 8.6

**WordPress Environment**
<details>
```

### WordPress Environment ###

WordPress address (URL): https://www.glamourshop.co.il/
Site address (URL): https://www.glamourshop.co.il/
WC Version: 6.3.0
REST API Version: ✔ 6.3.0
WC Blocks Version: ✔ 6.9.0
Action Scheduler Version: ✔ 3.4.0
WC Admin Version: ✔ 3.2.1
Log Directory Writable: ✔
WP Version: 5.9.1
WP Multisite: –
WP Memory Limit: 768 MB
WP Debug Mode: –
WP Cron: ✔
Language: he_IL
External object cache: ✔

### Server Environment ###

Server Info: Apache
PHP Version: 7.4.28
PHP Post Max Size: 256 MB
PHP Time Limit: 120
PHP Max Input Vars: 3000
cURL Version: 7.66.0
OpenSSL/1.1.1l-fips

SUHOSIN Installed: –
MySQL Version: 5.7.32-35-log
Max Upload Size: 256 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 6.3.0
WC Database Prefix: _VVX_
Total Database Size: 116.45MB
Database Data Size: 96.20MB
Database Index Size: 20.25MB
_VVX_woocommerce_sessions: Data: 9.58MB + Index: 0.21MB + Engine MyISAM
_VVX_woocommerce_api_keys: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_woocommerce_attribute_taxonomies: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_woocommerce_downloadable_product_permissions: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_woocommerce_order_items: Data: 0.32MB + Index: 0.14MB + Engine MyISAM
_VVX_woocommerce_order_itemmeta: Data: 24.44MB + Index: 1.61MB + Engine MyISAM
_VVX_woocommerce_tax_rates: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_woocommerce_tax_rate_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_woocommerce_shipping_zones: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_woocommerce_shipping_zone_locations: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_woocommerce_shipping_zone_methods: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_woocommerce_payment_tokens: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_woocommerce_payment_tokenmeta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_woocommerce_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_actionscheduler_actions: Data: 0.66MB + Index: 0.30MB + Engine MyISAM
_VVX_actionscheduler_claims: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_actionscheduler_groups: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
_VVX_actionscheduler_logs: Data: 0.51MB + Index: 0.39MB + Engine MyISAM
_VVX_adtribes_my_conversions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_affiliate_wp_affiliatemeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_affiliate_wp_affiliates: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_affiliate_wp_campaigns: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_affiliate_wp_coupons: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_affiliate_wp_creatives: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_affiliate_wp_customermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_affiliate_wp_customers: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_affiliate_wp_payouts: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_affiliate_wp_referralmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_affiliate_wp_referrals: Data: 0.08MB + Index: 0.02MB + Engine InnoDB
_VVX_affiliate_wp_rest_consumers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_affiliate_wp_sales: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_affiliate_wp_visits: Data: 1.52MB + Index: 0.23MB + Engine InnoDB
_VVX_commentmeta: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
_VVX_comments: Data: 0.61MB + Index: 0.19MB + Engine MyISAM
_VVX_db7_forms: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_ditty_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_ditty_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_icl_content_status: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_icl_core_status: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_icl_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_icl_languages: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_icl_languages_translations: Data: 0.19MB + Index: 0.09MB + Engine InnoDB
_VVX_icl_locale_map: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_icl_message_status: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_icl_mo_files_domains: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
_VVX_icl_node: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_icl_reminders: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_icl_strings: Data: 6.52MB + Index: 8.00MB + Engine InnoDB
_VVX_icl_string_batches: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_icl_string_packages: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_icl_string_pages: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_icl_string_positions: Data: 1.52MB + Index: 0.14MB + Engine InnoDB
_VVX_icl_string_status: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_icl_string_translations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_icl_string_urls: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_icl_translate: Data: 0.05MB + Index: 0.02MB + Engine InnoDB
_VVX_icl_translate_job: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_icl_translations: Data: 0.28MB + Index: 0.59MB + Engine InnoDB
_VVX_icl_translation_batches: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_icl_translation_downloads: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_icl_translation_status: Data: 0.19MB + Index: 0.05MB + Engine InnoDB
_VVX_links: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_mailchimp_carts: Data: 0.16MB + Index: 0.00MB + Engine InnoDB
_VVX_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_mailpoet_custom_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_dynamic_segment_filters: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_feature_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_forms: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_mailpoet_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_mailpoet_mapping_to_external_entities: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_newsletters: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
_VVX_mailpoet_newsletter_links: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
_VVX_mailpoet_newsletter_option: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_newsletter_option_fields: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_newsletter_posts: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_newsletter_segment: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_newsletter_templates: Data: 2.52MB + Index: 0.00MB + Engine InnoDB
_VVX_mailpoet_scheduled_tasks: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_mailpoet_scheduled_task_subscribers: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_segments: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_sending_queues: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_mailpoet_settings: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_statistics_clicks: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
_VVX_mailpoet_statistics_forms: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_statistics_newsletters: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_mailpoet_statistics_opens: Data: 0.02MB + Index: 0.08MB + Engine InnoDB
_VVX_mailpoet_statistics_unsubscribes: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
_VVX_mailpoet_statistics_woocommerce_purchases: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
_VVX_mailpoet_stats_notifications: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_mailpoet_subscribers: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
_VVX_mailpoet_subscriber_custom_field: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_subscriber_ips: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mailpoet_subscriber_segment: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_mailpoet_user_flags: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_mclean_refs: Data: 0.14MB + Index: 0.00MB + Engine InnoDB
_VVX_mclean_scan: Data: 0.11MB + Index: 0.02MB + Engine InnoDB
_VVX_news_announcement: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_options: Data: 10.76MB + Index: 0.20MB + Engine MyISAM
_VVX_postmeta: Data: 8.98MB + Index: 2.97MB + Engine MyISAM
_VVX_posts: Data: 2.39MB + Index: 0.34MB + Engine MyISAM
_VVX_redirection_404: Data: 1.02MB + Index: 0.05MB + Engine InnoDB
_VVX_redirection_groups: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_redirection_items: Data: 0.20MB + Index: 0.27MB + Engine InnoDB
_VVX_redirection_logs: Data: 2.02MB + Index: 0.20MB + Engine InnoDB
_VVX_sgs_log_events: Data: 13.02MB + Index: 0.00MB + Engine InnoDB
_VVX_sgs_log_visitors: Data: 0.17MB + Index: 0.00MB + Engine InnoDB
_VVX_smush_dir_images: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_social_users: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_termmeta: Data: 0.02MB + Index: 0.02MB + Engine MyISAM
_VVX_terms: Data: 0.01MB + Index: 0.04MB + Engine MyISAM
_VVX_term_relationships: Data: 0.09MB + Index: 0.17MB + Engine MyISAM
_VVX_term_taxonomy: Data: 0.02MB + Index: 0.02MB + Engine MyISAM
_VVX_usermeta: Data: 0.50MB + Index: 0.21MB + Engine MyISAM
_VVX_users: Data: 0.02MB + Index: 0.03MB + Engine MyISAM
_VVX_vtprd_purchase_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_vtprd_purchase_log_product: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_vtprd_purchase_log_product_rule: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_vtprd_transient_cart_data: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_wc_admin_notes: Data: 0.01MB + Index: 0.00MB + Engine MyISAM
_VVX_wc_admin_note_actions: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_wc_category_lookup: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_wc_customer_lookup: Data: 0.04MB + Index: 0.04MB + Engine MyISAM
_VVX_wc_download_log: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_wc_order_coupon_lookup: Data: 0.01MB + Index: 0.02MB + Engine MyISAM
_VVX_wc_order_product_lookup: Data: 0.35MB + Index: 0.28MB + Engine MyISAM
_VVX_wc_order_stats: Data: 0.04MB + Index: 0.04MB + Engine MyISAM
_VVX_wc_order_tax_lookup: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_wc_product_attributes_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_wc_product_meta_lookup: Data: 0.07MB + Index: 0.09MB + Engine MyISAM
_VVX_wc_rate_limits: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_wc_reserved_stock: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_wc_tax_rate_classes: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
_VVX_wc_webhooks: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_wdp_orders: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
_VVX_wdp_order_items: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
_VVX_wdp_rules: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_wdr_order_discounts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wdr_order_item_discounts: Data: 0.19MB + Index: 0.00MB + Engine InnoDB
_VVX_wdr_rules: Data: 0.05MB + Index: 0.00MB + Engine InnoDB
_VVX_wpam_actions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wpam_affiliates: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wpam_affiliates_fields: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wpam_creatives: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wpam_events: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wpam_impressions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wpam_messages: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wpam_paypal_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wpam_tracking_tokens: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wpam_tracking_tokens_purchase_logs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wpam_transactions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wpforms_tasks_meta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
_VVX_wpmailsmtp_debug_events: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_wpmailsmtp_tasks_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
_VVX_yith_wcwl: Data: 0.01MB + Index: 0.01MB + Engine MyISAM
_VVX_yith_wcwl_lists: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
_VVX_yoast_indexable: Data: 4.44MB + Index: 1.03MB + Engine InnoDB
_VVX_yoast_indexable_hierarchy: Data: 0.08MB + Index: 0.08MB + Engine InnoDB
_VVX_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
_VVX_yoast_primary_term: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
_VVX_yoast_seo_links: Data: 0.41MB + Index: 0.22MB + Engine InnoDB
_VVX_yoast_seo_meta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

acoplw_badges: 1
attachment: 1142
awl-labels: 1
blocks: 18
custom_css: 1
customize_changeset: 2
ditty_display: 3
ditty_layout: 3
ditty_news_ticker: 2
flamingo_contact: 168
flamingo_inbound: 128
jetpack_migration: 2
jp_img_sitemap: 1
jp_sitemap: 1
jp_sitemap_master: 1
mailpoet_page: 1
nav_menu_item: 55
oembed_cache: 69
page: 35
polylang_mo: 1
post: 12
product: 540
product_variation: 20
revision: 1
shop_coupon: 27
shop_order: 622
ux_template: 7
wcpa_pt_forms: 1
wp_global_styles: 1
wpcf7_contact_form: 5
wpcf7r_action: 10
wpcf7r_leads: 774

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (31) ###

Advanced Coupons for WooCommerce Free: by Rymera Web Co – 3.1.4
Advanced Woo Labels: by ILLID – 1.40
AffiliateWP: by Sandhills Development
LLC – 2.9.1

Click to Chat: by HoliThemes – 3.8
Contact Form 7 Multilingual: by OnTheGoSystems – 1.1.0
Contact Form 7: by Takayuki Miyoshi – 5.5.6
Ditty (formerly Ditty News Ticker): by Metaphor Creations – 3.0.16
Duplicate Page: by mndpsingh287 – 4.4.8
הנגשת אתרים enable: by uPress – 1.4
Jetpack: by Automattic – 10.7
Loco Translate: by Tim Whitlock – 2.6.0
Mailchimp for WooCommerce: by Mailchimp – 2.6.1
MC4WP: Mailchimp for WordPress: by ibericode – 4.8.7
Woocommerce Baldar Delivery: by Amit Matat – 1.5.0
Redirection: by John Godley – 5.2.3
SiteGround Optimizer: by SiteGround – 7.0.6
SiteGround Security: by SiteGround – 1.2.1
WooCommerce Custom Product Addons (Free): by Acowebs – 2.6.0
Woo Discount Rules PRO 2.0: by Flycart – 2.3.13
Woo Discount Rules: by Flycart – 2.3.13
Woo Image SEO: by Danail Emandiev – 1.2.5
WooCommerce Payment Gateway - SUMIT: by OfficeGuy Ltd. – 3.0.2
Product Feed PRO for WooCommerce: by AdTribes.io – 11.3.8
WooCommerce Google Analytics Integration: by WooCommerce – 1.5.8
More Sorting Options for WooCommerce: by WP Wham – 3.2.8
WooCommerce Smart Coupons: by StoreApps – 4.38.0
WooCommerce: by Automattic – 6.3.0
Yoast SEO: by Team Yoast – 18.3
WP Mail SMTP: by WPForms – 3.3.0
Redirection for Contact Form 7: by Qube One – 2.6.0
YITH WooCommerce Wishlist: by YITH – 3.7.0

### Inactive Plugins (0) ###

### Dropin Plugins (1) ###

object-cache.php: Memcached

### Settings ###

API Enabled: –
Force SSL: –
Currency: ILS (₪)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 0
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: [#65](65-zd-woothemes) - /shop/
Cart: [#64](64-zd-woothemes) - /shop/cart/
Checkout: [#5673](5673-zd-woothemes) - /shop/checkout/
My account: [#5674](5674-zd-woothemes) - /my-account/
Terms and conditions: [#3](3-zd-woothemes) - /privacy-policy/

### Theme ###

Name: Flatsome Child
Version: 3.0
Author URL:
Child Theme: ✔
Parent Theme Name: Flatsome
Parent Theme Version: 3.15.3
Parent Theme Author URL: http://www.uxthemes.com/
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart-shipping.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome/woocommerce/checkout/terms.php
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product-cat.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/content-widget-product.php
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/rating.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php
flatsome/woocommerce/content-product_cat.php

### More Sorting Settings ###

More Sorting for WooCommerce: yes
Custom Sorting: yes
Sort by date (ascending):
Sort by date (descending):
Sort by title: A to Z: למיין לפי שם (א'-ת')
Sort by title: Z to A: למיין לפי שם (ת'-א')
Sort by slug: A to Z:
Sort by slug: Z to A:
Sort by SKU: low to high:
Sort by SKU: high to low:
Sort by SKU: no
Sort by stock quantity: low to high:
Sort by stock quantity: high to low:
Sort by total sales: low to high:
Sort by total sales: high to low:
Sort by last modified date: oldest to newest:
Sort by last modified date: newest to oldest:
Sort by author ID (ascending):
Sort by author ID (descending):
Sort by product ID (ascending):
Sort by product ID (descending):
Sort by number of comments (ascending):
Sort by number of comments (descending):
Random sorting:
No sorting:
Custom Meta Sorting: no
Total Options: 1
Enable/Disable: yes
Type: meta_value
Meta Key:
Secondary Sorting: none
Order: asc
Parameter: custom_sorting_1
Title: Custom Meta Sorting [#1](1-zd-woothemes)
Default Sorting Options: no
Default sorting: Default sorting
Sort by popularity: Sort by popularity
Remove "Sort by popularity": no
Sort by average rating: Sort by average rating
Remove "Sort by average rating": no
Sort by newness: Sort by newness
Remove "Sort by newness": no
Sort by price: low to high: Sort by price: low to high
Remove "Sort by price: low to high": no
Sort by price: high to low: Sort by price: high to low
Remove "Sort by price: high to low": no
Rearrange Sorting: no
Rearrange Sorting: menu_order
popularity
rating
date
price
price-desc
date-asc
date-desc
title-asc
title-desc
name-asc
name-desc
sku-asc
sku-desc
stock_quantity-asc
stock_quantity-desc
total_sales-asc
total_sales-desc
modified-asc
modified-desc
author-asc
author-desc
product_id-asc
product_id-desc
comment_count-asc
comment_count-desc
rand
none
Remove All Sorting (Including WooCommerce Default): no
Enable/Disable: no
Theme: avada

### Action Scheduler ###

Canceled: 3
Oldest: 2022-02-11 10:56:48 +0200
Newest: 2022-03-06 20:09:54 +0200

Complete: 2,549
Oldest: 2022-02-10 16:52:28 +0200
Newest: 2022-03-09 20:02:41 +0200

Pending: 1
Oldest: 2022-03-14 16:16:29 +0200
Newest: 2022-03-14 16:16:29 +0200

### Status report information ###

Generated at: 2022-03-09 20:14:08 +02:00
```
</details>

**Comment by anitaa1990** [comment]
cc @ThomazFB @hichamboushaba @wzieba , this issue seems related to the OrderDAO changes. Would one of you mind taking a look this week please?

**Comment by ThomazFB** [comment]
I've also noticed in sentry an [issue similar to this one](https://sentry.io/organizations/a8c/issues/3090553318/?project=1459556), but instead of null line items, it's null metadata. @wzieba should we make the nullability of the Order model fields more flexible?

**Comment by wzieba** [comment]
Yes, as API contract is not as stable as we would wish (is that because of plugins? 🤷 ) we should make our restrictions less conservative. 

I wonder why it didn't popped out before, when using WellSql. I hope to learn it in the proccess.

Linking recent discussion about nullable fields in `OrderEntity`: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2295#discussion_r810347906

**Comment by hichamboushaba** [comment]
> I wonder why it didn't popped out before, when using WellSql. I hope to learn it in the proccess.

Maybe it has to do with how Room will map the `"null"` string vs what WellSql was doing, currently we map the API response fields using `toString` in our [mapper](https://github.com/wordpress-mobile/WordPress-FluxC-Android/blob/eca450e0f2ce35b636eee92b21632d5f7bdae548/plugins/woocommerce/src/main/kotlin/org/wordpress/android/fluxc/network/rest/wpcom/wc/order/OrderDto.kt#L132-L136), which would result in saving the String value `"null"` to the DB entity.

**Comment by wzieba** [comment]
@hichamboushaba that's good idea but I couldn't reproduce it in any way, on different phones.

The issues from Sentry shows that this happens on Samsung phones so your theory might be very true - maybe SQL driver on some Samsung phones treats `"null"` as `NULL`. 

And honestly, saving String with value `"null"` instead of empty String is a bug in our code already I think.

Applying the same logic of `?: ""` or `.orEmpty()`, like in other fields, would probably fix this issue but I'll try to reproduce the issue with help of other teammates that have Samsung phones with this tests: https://github.com/wordpress-mobile/WordPress-FluxC-Android/compare/woo/6013_add_test_cases_for_db_null_value?expand=1

**Comment by hichamboushaba** [comment]
> And honestly, saving String with value "null" instead of empty String is a bug in our code already I think.

Yes, I agree it's a bug, and I don't think it was intentional, it was just missed in the mapper.

**Comment by wzieba** [comment]
Just realised that this has been resolved "accidentally" by https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2308 PR - now, the null will be mapped to empty JSON array which is even more accurate than empty String I believe (as those fields are meant to be JSONs):

<img width="288" alt="image" src="https://user-images.githubusercontent.com/5845095/158802397-8a25ce8f-f525-4ff5-8af8-a56fcbb64d28.png">

---

As I wasn't able to reproduce the issue, I close it on Sentry (mark as resolved in `8.8`) and we'll observe if it'll return.

Edit: I'll do this ☝️ when `8.8` will be available in Sentry.

**Comment by thabotswana** [comment]
The user had asked for an update. Is there anything we can tell them?

4844740-zen

**Comment by wzieba** [comment]
@thabotswana I'm now trying to reproduce the issue once again and check if recent changes fixed the thing. I should have some answers EOD, I'll ping you here. Is that okay?

**Comment by wzieba** [comment]
@thabotswana unfortunately, I haven't found a way to reproduce the original issue so far. We can just hope that `8.8` release will resolve the issue. If not, we'll continue working on it.

**Comment by erricgunawan** [comment]
@wzieba just to confirm, does this mean we should ask the user to wait until the `8.8` is released? 

Or do y'all still want to run some testing or continue working on this?

**Comment by wzieba** [comment]
> @wzieba just to confirm, does this mean we should ask the user to wait until the 8.8 is released?

Yes, we can communicate that `8.8` is _supposed_ to fix the bug. If they'll still report the issue, we'll work on that further. 

Thanks and sorry for that uncertainty - as I couldn't reproduce issue, we can't be sure that we have the fix.

**Comment by wzieba** [comment]
I've looked at Sentry dashboard and it seems that we successfully fixed this bug in `8.8`: https://sentry.io/organizations/a8c/discover/results/?id=13523&project=1459556&statsPeriod=90d

Closing this issue then.
